src/core.jl

```diff
@@ -73,12 +73,14 @@ function issubset{T<:SimpleGraph}(g::T, h::T)
 end
 
 
-"""Add `n` new vertices to the graph `g`."""
+"""Add `n` new vertices to the graph `g`. Returns true if all vertices
+were added successfully, false otherwise."""
 function add_vertices!(g::SimpleGraph, n::Integer)
+    added = true
     for i = 1:n
-        add_vertex!(g)
+        added &= add_vertex!(g)
```

Comment:
Yes - that was my thinking. In the event we ever change the return logic for `add_vertex!`, we shouldn't need to change anything else.